pkg/sfu/buffer/buffer.go

```diff
@@ -408,37 +404,24 @@ func (b *Buffer) updateStreamState(p *rtp.Packet, arrivalTime int64) RTPFlowStat
 func (b *Buffer) processHeaderExtensions(p *rtp.Packet, arrivalTime int64) {
 	// submit to TWCC even if it is a padding only packet. Clients use padding only packets as probes
 	// for bandwidth estimation
-	if b.twcc {
-		if ext := p.GetExtension(b.twccExt); len(ext) > 1 {
-			b.twccPending = append(b.twccPending, twccMetadata{
-				sn:          binary.BigEndian.Uint16(ext[0:2]),
-				arrivalTime: arrivalTime,
-				marker:      p.Marker,
-			})
-			if len(b.twccPending) == 1 {
-				b.callbacksQueue.Enqueue(b.processTWCCPending)
-			}
+	if b.twcc != nil && b.twccExt != 0 {
+		if ext := p.GetExtension(b.twccExt); ext != nil {
+			b.twcc.Push(binary.BigEndian.Uint16(ext[0:2]), arrivalTime, p.Marker)
 		}
 	}
 
-	if b.audioLevel {
+	if b.audioLevelExt != 0 {
 		if !b.latestTSForAudioLevelInitialized {
 			b.latestTSForAudioLevelInitialized = true
 			b.latestTSForAudioLevel = p.Timestamp
 		}
```

Comment:
Plan to move this kind of state into audio level. Thinking about making a different flavour of active speaker detection. Will think about moving these at that time after seeing how that interface will work out.